compiler/noirc_evaluator/src/ssa/acir_gen/acir_ir/generated_acir.rs

```diff
@@ -646,6 +646,10 @@ impl GeneratedAcir {
         self.push_opcode(AcirOpcode::Arithmetic(expr));
     }
 
+    pub(crate) fn assert_eq(&mut self, lhs: Expression, rhs: Expression) {
+        self.push_opcode(AcirOpcode::AssertEq { lhs, rhs });
+    }
```

Comment:
I think this is a bit of a footgun as I would expect that an `AssertEq` opcode would lay down constraints which assert that the two arguments are equal.

Reply:
The assert is the difference of the two expressions. If that does not give 0, then there is no way to get the values of lhs and rhs from their difference. So we need this extra information.